internal/kuberesource/parts.go

```diff
@@ -204,6 +207,20 @@ func NodeInstaller(namespace string, platform platforms.Platform) (*NodeInstalle
 	}, nil
 }
 
+// NodeInstallerTargetConfig returns a ConfigMap for the passed target.
+func NodeInstallerTargetConfig(target, namespace string) (*applycorev1.ConfigMapApplyConfiguration, error) {
+	switch target {
+	case "k3s":
+		return applycorev1.ConfigMap("contrast-node-installer-target-config", namespace).
+			WithData(map[string]string{
+				"containerd-config-path": "var/lib/rancher/k3s/agent/etc/containerd/config.toml.tmpl",
+				"systemd-unit-name":      "k3s.service,k3s-agent.service",
+			}), nil
+	default:
+		return nil, fmt.Errorf("unsupported target %q", target)
```

Comment:
IMO it would be nice to give indication that the target (platform) is either unsupported, or that it is not required to be specified. There's probably not a good way to decide which is the case here, but I could easily see someone passing "metal" or the like here and being confused.

Reply:
I think that should be handled by the caller.